src/compiler/scala/tools/nsc/transform/Constructors.scala

```diff
@@ -691,14 +690,22 @@ abstract class Constructors extends Statics with Transform with TypingTransforme
 
             case dd: DefDef =>
               // either move the RHS to ctor (for getter of stored field) or just drop it (for corresponding setter)
-              def shouldMoveRHS =
-                clazz.isTrait && statSym.isAccessor && !statSym.isLazy && !statSym.isSpecialized && (statSym.isSetter || memoizeValue(statSym))
-
-              if ((dd.rhs eq EmptyTree) || !shouldMoveRHS) { defBuf += dd }
-              else {
-                if (statSym.isGetter) moveEffectToCtor(dd.mods, dd.rhs, statSym.asTerm.referenced orElse statSym.setterIn(clazz))
-                defBuf += deriveDefDef(stat)(_ => EmptyTree)
-              }
+              def shouldMoveRHS = (
+                   (dd.rhs ne EmptyTree)
+                && clazz.isTrait
+                && statSym.isAccessor
+                && !statSym.isLazy
+                && !statSym.isSpecialized
+                && (statSym.isSetter || memoizeValue(statSym))
+              )
+              val toMove =
+                if (shouldMoveRHS) {
+                  if (statSym.isGetter)
+                    moveEffectToCtor(dd.mods, dd.rhs, statSym.asTerm.referenced.orElse(statSym.setterIn(clazz)))
+                  deriveDefDef(stat)(_ => EmptyTree)
+                }
+                else dd
+              defBuf += toMove
```

Comment:
I used to think vertical boolean exprs (wrapped in parens) were too much style just to have leading infix operators, but after using actual leading infix (without parens), I find it much easier to read. I only preferred the one-liner because mentally I would skip it and that's why it felt easier to read.

note that under `-Xsource:3` we could be using leading infix.